lib/spack/spack/cmd/diff.py

```diff
@@ -173,17 +161,17 @@ def group_by_type(diffset):
             category = key
 
             # print category in bold, colorized
-            out.write(boldblue("@@ %s @@\n" % category))
+            cprint("@*b{@@ %s @@}" % category)
 
         # Print subtractions first
         while subtraction:
-            out.write(red("-  %s\n" % subtraction.pop(0)))
+            cprint("@R{-  %s}" % subtraction.pop(0))
```

Comment:
The reason I had those functions named red, boldblue, was for understandability. So if you are removing them could you please put comments above the first instance of these lines or in the function docstring to say what the format strings do (e.g bold blue) a new contributor won’t know.

Reply:
We use the color stuff all over spack and the letters are meant to be mnemonic... for readability.  Admittedly, it is not 100% intuitive but it is better than actual ANSI color escapes like `\033[91m`.

Would it help to make the color library use names instead of letters?  That seems like something for another PR...

Reply:
That could help, but for now I think we should add the docstring so the developer user doesn’t need to go searching around spack to understand it. I already went through that and (until there is some update so it’s easy to parse) I don’t want someone else to have to go through that random grepping and testing to understand a new syntax.

Reply:
don